client/common/zod.js

```diff
@@ -0,0 +1,29 @@
+import z from 'zod';
+
+/* eslint-disable */
+const EMAIL_REGEX = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/i;
+/* eslint-enable */
+
+const SignupFormInput = z.object({
+  username: z
+    .string()
+    .min(1)
+    .max(20)
+    .regex(/^[a-zA-Z0-9._-]{1,20}$/),
+  email: z.string().email().regex(EMAIL_REGEX),
```

Comment:
I wonder if the Zod .email() function checks for the same stuff as our EMAIL_REGEX?  It might be that we can delete the regex check — but I need to check the Zod docs and/or source code to know.

Reply:
I confirmed that Zod has their own email regex: https://github.com/colinhacks/zod/blob/a5a9d31018f9c27000461529c582c50ade2d3937/src/types.ts#L567-L568

Reply:
@lindapaiste so should I use the zod email regex or this one is fine